app/services/katello/pulp3/migration_plan.rb

```diff
@@ -105,7 +105,15 @@ def target_link_map
       end
 
       def name_for_content_view(content_view, root_repo)
-        "#{content_view.label}-#{root_repo.label}"
+        name = "#{content_view.label}-#{root_repo.label}"
+
+        if Katello::RootRepository.where(:label => root_repo.label).group(:label).count(:label)[root_repo.label] > 1
```

Comment:
I don't really know ruby, but doesn't that translate to the SQL (root_repo.label being 2_3_el7_x86_64)

`select count(label) from katello_root_repositories where label = '2_3_el7_x86_64' group by label;`

(I don't understand what the `[root_repo.label]` does after that)

The group seems unnecessary here as all you want to do is to count the number of rows with the specific root_repo.label, i.e. in the end

`select count(*) from katello_root_repositories where label = '2_3_el7_x86_64';`

With my nonexisting ruby knowledge, that should be something like

`Katello::RootRepository.where(:label => root_repo.label).count`

?

Reply:
oh i think you're right! thats overly complicated

Reply:
I think that should still simply be 
```
if Katello::RootRepository.where(:label => root_repo.label).count > 1
```

Due to the where, there is always only a single group, making the whole group part unnecessary...